test/src/test/java/hudson/ProcTest.java

```diff
@@ -57,14 +57,15 @@ public void run() {
 
         RemoteLauncher launcher = new RemoteLauncher(StreamTaskListener.NULL, ch, true);
 
-        String str="";
+        StringBuilder stringBuilder = new StringBuilder();
+
         for (int i=0; i<256; i++)
-            str += "oxox";
+            stringBuilder.append("oxox");
 
         for (int i=0; i<1000; i++) {
             ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            launcher.launch().cmds("echo",str).stdout(baos).join();
-            assertEquals(str, baos.toString().trim());
+            launcher.launch().cmds("echo",stringBuilder.toString()).stdout(baos).join();
+            assertEquals(stringBuilder, baos.toString().trim());
```

Comment:
It causes a test failure:

<img width="1013" alt="screen shot 2017-03-17 at 14 07 31" src="https://cloud.githubusercontent.com/assets/3000480/24044338/279694ee-0b1b-11e7-8d65-e08f77d91b29.png">